server/src/routes/donatedItemStatusRoutes.ts

```diff
@@ -0,0 +1,35 @@
+import { Router, Request, Response } from 'express';
+import prisma from '../prismaClient'; // Import Prisma client
+import { donatedItemStatusValidator } from '../validators/donatedItemStatusValidator'; // Import the validator
+
+const router = Router();
+
+// PUT /donatedItem/status/:id - Update the status of a DonatedItem
+router.put('/:id', donatedItemStatusValidator, async (req: Request, res: Response) => {
+    try {
+        const updatedStatus = await prisma.donatedItem.update({
+            where: { id: Number(req.params.id) },
+            data: {
+                currentStatus: req.body.statusType,
+                lastUpdated: new Date(),
+            },
+        });
+
+        // Create a new entry in DonatedItemStatus to track the status change
+        const newStatus = await prisma.donatedItemStatus.create({
+            data: {
+                statusType: req.body.statusType,
+                dateModified: new Date(),
+                donatedItemId: Number(req.params.id),
```

Comment:
Add a check to avoid updating if the new status is the same as the existing one. This helps reduce unwanted updates and unnecessary status history entries.